hca-dcp/submit.wdl

```diff
@@ -0,0 +1,163 @@
+      --compressed \
+      "$(cut -f3 $creds)/api/workflows/v1/$(cat workflow_id.txt)/metadata" > metadata.json
+  >>>
+  runtime {
+    docker: "gcr.io/broad-dsde-mint-dev/cromwell-metadata:test1"
+  }
+  output {
+    File metadata = "metadata.json"
+    String workflow_id = read_string("workflow_id.txt")
+  }
+}
+
+# Create the submission object in ingest
+task create_submission {
+  String workflow_id
+  File metadata_json
+  String input_bundle_uuid
+  String reference_bundle
+  String run_type
+  String method
+  String schema_version
+  Array[Object] inputs
+  Array[String] outputs
+  File format_map
+  String submit_url
+
+  command <<<
+    # First, create the analysis.json
+    # Note that create-analysis-json can take a comma-separated list of bundles,
+    # but current workflows only take a single input bundle
+    create-analysis-json \
+      -analysis_id ${workflow_id} \
+      -metadata_json ${metadata_json} \
+      -input_bundles ${input_bundle_uuid} \
+      -reference_bundle ${reference_bundle} \
+      -run_type ${run_type} \
+      -method ${method} \
+      -schema_version ${schema_version} \
+      -inputs_file ${write_objects(inputs)} \
+      -outputs_file ${write_lines(outputs)} \
+      -format_map ${format_map}
+
+    # Now create the submission object
+    create-envelope \
+      -submit_url ${submit_url} \
+      -analysis_json_path analysis.json
+  >>>
+
+  runtime {
+    docker: "humancellatlas/secondary-analysis-python:test16"
+  }
+  output {
+    File analysis_json = "analysis.json"
+    String submission_url = read_string("submission_url.txt")
+  }
+}
+
+# Stage files, then confirm submission
+task stage_and_confirm {
+  String submission_url
+  Array[File] files
+  Int retry_seconds
+  Int timeout_seconds
+  # These left and right brace definitions are a workaround so Cromwell won't
+  # interpret the bash array reference below as a WDL variable.
+  String lb = "{"
+  String rb = "}"
+
+  command <<<
+    set -e
+
+    # Get the urn needed for staging files
+    staging_urn=$(get-staging-urn \
+        -envelope_url ${submission_url} \
+        -retry_seconds ${retry_seconds} \
+        -timeout_seconds ${timeout_seconds})
+
+    # Stage the files
+    files=( ${sep=' ' files} )
+    for f in "$${lb}files[@]${rb}"
+    do
+      echo "stage $f $staging_urn"
+      stage $f $staging_urn
+    done
+
+    # Confirm the submission
+    confirm-submission \
+      -envelope_url ${submission_url} \
+      -retry_seconds ${retry_seconds} \
+      -timeout_seconds ${timeout_seconds}
+  >>>
+
+  runtime {
+    docker: "humancellatlas/secondary-analysis-python:test16"
+  }
+}
+
+workflow submit {
+  Array[Object] inputs
+  Array[File] outputs
+  File format_map
+  String submit_url
+  String input_bundle_uuid
+  String reference_bundle
+  String run_type
+  String schema_version
+  String method
+  Int retry_seconds
+  Int timeout_seconds
+
+  call get_metadata {
+    input:
+      analysis_output_path = outputs[0]
```

Comment:
The `analysis_output_path` is referring to the `bam_file` from the analysis workflow, right? I think it would be clearer to pass it in as a separate input instead of making sure it's always the first item in `outputs`.

Reply:
Hmm, let's talk about this in person. Basically we need any file output path, purely for the sake of parsing the workflow id from it.

Reply:
Ok, I see what you mean now!